ouroboros-consensus/ouroboros-consensus-test-infra/src/Test/Util/Pam.hs

```diff
@@ -0,0 +1,60 @@
+module Test.Util.Pam (
+  -- * Pam type
+  Pam,
+  -- * Query
+  Test.Util.Pam.null,
+  -- * Construction
+  fromMap,
+  unsafePam,
+  -- * Conversion
+  getPam,
+  toMap,
+  unsafeCoercion,
+  -- * Filter
+  spanAntitone,
+  -- * Min/Max
+  minViewWithKey,
+  ) where
+
+import           Data.List.NonEmpty (NonEmpty)
+import qualified Data.List.NonEmpty as NE
+import           Data.Map (Map)
+import qualified Data.Map as Map
+import           Data.Type.Coercion
+
+-- | An inverted 'Map'
+--
+-- INVARIANT the @k@s are all unique
+--
+-- INVARIANT the 'NonEmpty's are all ascending
+--
+newtype Pam v k = UnsafePam {getPam :: Map v (NonEmpty k)}
```

Comment:
Let's rename this to `InvertedMap`.

Reply:
Done in commit 3de1fc8401ba1a57fbe875fba9180217a1f98c92